tests/unit/functions.tcl

```diff
@@ -117,6 +117,22 @@ start_server {tags {"scripting"}} {
         r fcall test 0
     } {hello}
 
+    test {FUNCTION - test debug reload different options} {
+        catch {r debug reload noflush} e
+        assert_match "*Error trying to load the RDB*" $e
+        r debug reload noflush merge
+        r function list
+    } {{name test engine LUA description {some description}}}
+
+    test {FUNCTION - test debug reload with nosave and noflush} {
+        r function delete test
+        r set x 1
+        r debug reload
+        r function create LUA test DESCRIPTION {some description} {return 'hello'}
+        r debug reload nosave noflush merge
+        r function list
+    } {{name test engine LUA description {some description}}}
```

Comment:
don't we want to add another function so that the "merge" option has the opportunity to end up with two functions, each from the different source?

Reply:
Added another function before the first debug reload